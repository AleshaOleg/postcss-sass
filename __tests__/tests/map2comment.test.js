var getPostCssTreeFromSass = require('../../helpers/getPostCssTreeFromSassTree');

it('map2comment.sass', function () {
    expect(getPostCssTreeFromSass('map2comment')).toMatchSnapshot();
});
