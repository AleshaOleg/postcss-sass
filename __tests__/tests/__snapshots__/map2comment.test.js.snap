// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map2comment.sass 1`] = `
Object {
  "nodes": Array [
    Object {
      "raws": Object {
        "before": "",
        "commentType": "multi",
        "left": " ",
        "right": " ",
      },
      "text": "Comment",
      "type": "comment",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "red",
          "raws": Object {
            "before": "
  ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 15,
              "line": 3,
            },
            "input": Input {
              "css": "/* Comment */
$button-colors:
  red: #e74c3c,
  blue: #3498db,
  green: #2ecc71,

// Comment
@each $name, $color in $button-colors
  .button-#{$name}
    background-color: $color
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 3,
            },
          },
          "type": "decl",
          "value": "#e74c3c,",
        },
        Object {
          "prop": "blue",
          "raws": Object {
            "before": "
  ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 16,
              "line": 4,
            },
            "input": Input {
              "css": "/* Comment */
$button-colors:
  red: #e74c3c,
  blue: #3498db,
  green: #2ecc71,

// Comment
@each $name, $color in $button-colors
  .button-#{$name}
    background-color: $color
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 4,
            },
          },
          "type": "decl",
          "value": "#3498db,",
        },
        Object {
          "prop": "green",
          "raws": Object {
            "before": "
  ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 17,
              "line": 5,
            },
            "input": Input {
              "css": "/* Comment */
$button-colors:
  red: #e74c3c,
  blue: #3498db,
  green: #2ecc71,

// Comment
@each $name, $color in $button-colors
  .button-#{$name}
    background-color: $color
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 5,
            },
          },
          "type": "decl",
          "value": "#2ecc71,",
        },
      ],
      "raws": Object {
        "before": "
",
        "between": "",
      },
      "selector": "$button-colors:",
      "source": Object {
        "end": Object {
          "column": 18,
          "line": 5,
        },
        "input": Input {
          "css": "/* Comment */
$button-colors:
  red: #e74c3c,
  blue: #3498db,
  green: #2ecc71,

// Comment
@each $name, $color in $button-colors
  .button-#{$name}
    background-color: $color
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 2,
        },
      },
      "type": "rule",
    },
    Object {
      "raws": Object {
        "before": "

",
        "commentType": "single",
        "left": " ",
        "right": "",
      },
      "text": "Comment",
      "type": "comment",
    },
    Object {
      "nodes": Array [
        Object {
          "nodes": Array [
            Object {
              "prop": "background-color",
              "raws": Object {
                "before": "
    ",
                "between": ": ",
                "semicolon": false,
              },
              "source": Object {
                "end": Object {
                  "column": 28,
                  "line": 10,
                },
                "input": Input {
                  "css": "/* Comment */
$button-colors:
  red: #e74c3c,
  blue: #3498db,
  green: #2ecc71,

// Comment
@each $name, $color in $button-colors
  .button-#{$name}
    background-color: $color
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 5,
                  "line": 10,
                },
              },
              "type": "decl",
              "value": "$color",
            },
          ],
          "raws": Object {
            "before": "
  ",
            "between": "",
          },
          "selector": ".button-#{name}",
          "source": Object {
            "end": Object {
              "column": 29,
              "line": 10,
            },
            "input": Input {
              "css": "/* Comment */
$button-colors:
  red: #e74c3c,
  blue: #3498db,
  green: #2ecc71,

// Comment
@each $name, $color in $button-colors
  .button-#{$name}
    background-color: $color
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 9,
            },
          },
          "type": "rule",
        },
      ],
      "raws": Object {
        "before": "
",
        "between": "",
      },
      "selector": "@each $name, $color in $button-colors",
      "type": "rule",
    },
  ],
  "raws": Object {
    "before": "",
    "semicolon": undefined,
  },
  "source": Object {
    "end": Object {
      "column": 29,
      "line": 10,
    },
    "input": Input {
      "css": "/* Comment */
$button-colors:
  red: #e74c3c,
  blue: #3498db,
  green: #2ecc71,

// Comment
@each $name, $color in $button-colors
  .button-#{$name}
    background-color: $color
",
      "id": "<input css 1>",
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "type": "root",
}
`;
